provider/server/status_test.go

```diff
@@ -0,0 +1,52 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package server_test
+
+import (
+	iriv1alpha1 "github.com/ironcore-dev/ironcore/iri/apis/machine/v1alpha1"
+	"github.com/ironcore-dev/libvirt-provider/pkg/mcr"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Status", func() {
+	It("should get list of supported machine class with calculated quantity in status", func(ctx SpecContext) {
+		By("getting machine class status")
+		statusResp, err := machineClient.Status(ctx, &iriv1alpha1.StatusRequest{})
+		Expect(err).NotTo(HaveOccurred())
+
+		By("loading machine classes from file")
+		machineClasses, err := mcr.LoadMachineClasses(machineClassesFile)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("getting host resources")
+		hostResources, err := mcr.GetResources(ctx, false)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("validating machine class and calculated quantity in MachineClassStatus")
+		Expect(statusResp.MachineClassStatus).To(ContainElements(
```

Comment:
optional: ConsistOf function is more restrict and it can help find more problems in future

Reply:
If we use ConsistOf will have to check other fields like XXX_NoUnkeyedLiteral and XXX_sizecache unnecessarily